### PR TITLE
Parse links from Slack message text

### PR DESCRIPTION
**Pull request by leogrim**
@ryanpbrewster

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/model/KeepInfo.scala [hunk 1]
```diff
@@ -81,6 +81,6 @@ object KeepInfo {
   // Are you looking for a decorated keep (with tags, rekeepers, etc)?
   // Use KeepsCommander#decorateKeepsIntoKeepInfos(userId, keeps)
   def fromKeep(bookmark: Keep)(implicit publicIdConfig: PublicIdConfiguration): KeepInfo = {
-    KeepInfo(Some(bookmark.externalId), bookmark.title, bookmark.url, bookmark.isPrivate, user = None, libraryId = bookmark.libraryId.map(Library.publicId))
+    KeepInfo(Some(bookmark.externalId), bookmark.title, bookmark.url, bookmark.isPrivate, user = None, libraryId = bookmark.libraryId.map(Library.publicId), sourceAttribution = None)
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/model/KeepInfo.scala:84]: Isn't this wrong?

Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/model/KeepInfo.scala:84]: It was implicitly wrong, now it's explicitly wrong.

File: kifi-backend/modules/shoebox/app/com/keepit/slack/SlackIngestionCommander.scala [hunk 2]
```diff
@@ -122,13 +124,28 @@ class SlackIngestionCommanderImpl @Inject() (
     lastMessageTimestamp
   }
 
-  private def toRawBookmarks(message: SlackMessage): Seq[RawBookmarkRepresentation] = {
-    if (message.userId.value.trim.isEmpty || SlackUsername.doNotIngest.contains(message.username)) Seq.empty[RawBookmarkRepresentation]
-    else message.attachments.flatMap { attachment => // todo(Léo): messages can have links but no attachment
-      (attachment.title.flatMap(_.link) orElse attachment.fromUrl).collect {
+  private def doNotIngest(message: SlackMessage): Boolean = message.userId.value.trim.isEmpty || SlackUsername.doNotIngest.contains(message.username)
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/slack/SlackIngestionCommander.scala:127]: Can we use a different name for the method `SlackMessage => Boolean` and the set `Set[SlackUsername]`?

Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/slack/SlackIngestionCommander.scala:127]: No.

**Pull request merged**